From e19da7c3dc86c6f937dc2a43d0afbff8ecec933a Mon Sep 17 00:00:00 2001
From: Simon Horman <horms+renesas@verge.net.au>
Date: Wed, 3 Jul 2013 13:38:22 +0900
Subject: [PATCH 0947/1083] Revert "drm/rcar-du: Add XRGB1555 format support"

This reverts commit 25eebe006ca30a5333ab5a81faf6a13e143caba7.
(cherry picked from commit 0b0846001059d823c7b7fa17c365b74e2ef22081)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_kms.c   | 6 ------
 drivers/gpu/drm/rcar-du/rcar_du_plane.c | 2 --
 2 files changed, 8 deletions(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_kms.c b/drivers/gpu/drm/rcar-du/rcar_du_kms.c
index 873bb23..1cc27bb 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_kms.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_kms.c
@@ -42,12 +42,6 @@ static const struct rcar_du_format_info rcar_du_format_infos[] = {
 		.pnmr = PnMR_SPIM_ALP | PnMR_DDDF_ARGB,
 		.edf = PnDDCR4_EDF_NONE,
 	}, {
-		.fourcc = DRM_FORMAT_XRGB1555,
-		.bpp = 16,
-		.planes = 1,
-		.pnmr = PnMR_SPIM_TP | PnMR_DDDF_ARGB,
-		.edf = PnDDCR4_EDF_NONE,
-	}, {
 		.fourcc = DRM_FORMAT_XRGB8888,
 		.bpp = 32,
 		.planes = 1,
diff --git a/drivers/gpu/drm/rcar-du/rcar_du_plane.c b/drivers/gpu/drm/rcar-du/rcar_du_plane.c
index 0a30f89..8cbd293 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_plane.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_plane.c
@@ -161,7 +161,6 @@ static void rcar_du_plane_setup_mode(struct rcar_du_plane *plane,
 		break;
 
 	case DRM_FORMAT_ARGB1555:
-	case DRM_FORMAT_XRGB1555:
 		colorkey = ((plane->colorkey & 0xf80000) >> 9)
 			 | ((plane->colorkey & 0x00f800) >> 6)
 			 | ((plane->colorkey & 0x0000f8) >> 3);
@@ -391,7 +390,6 @@ static const struct drm_plane_funcs rcar_du_plane_funcs = {
 static const uint32_t formats[] = {
 	DRM_FORMAT_RGB565,
 	DRM_FORMAT_ARGB1555,
-	DRM_FORMAT_XRGB1555,
 	DRM_FORMAT_XRGB8888,
 	DRM_FORMAT_ARGB8888,
 	DRM_FORMAT_UYVY,
-- 
1.8.3.2

